python/ray/tests/test_logging.py

```diff
@@ -138,52 +156,58 @@ def test_log_file_exists(shutdown_only):
     session_path = Path(session_dir)
     log_dir_path = session_path / "logs"
 
-    # NOTICE: There's no ray_constants.PROCESS_TYPE_WORKER because "worker" is a
-    # substring of "python-core-worker".
-    log_rotating_component = [
-        (ray_constants.PROCESS_TYPE_DASHBOARD, [".log", ".out", ".err"]),
-        (ray_constants.PROCESS_TYPE_DASHBOARD_AGENT, [".log"]),
-        (ray_constants.PROCESS_TYPE_RUNTIME_ENV_AGENT, [".log", ".out", ".err"]),
-        (ray_constants.PROCESS_TYPE_LOG_MONITOR, [".log", ".err"]),
-        (ray_constants.PROCESS_TYPE_MONITOR, [".log", ".out", ".err"]),
-        (ray_constants.PROCESS_TYPE_PYTHON_CORE_WORKER_DRIVER, [".log"]),
-        (ray_constants.PROCESS_TYPE_PYTHON_CORE_WORKER, [".log"]),
-        (ray_constants.PROCESS_TYPE_RAYLET, [".out", ".err"]),
-        (ray_constants.PROCESS_TYPE_GCS_SERVER, [".out", ".err"]),
-    ]
+    # Run a no-op task to ensure all logs are created.
+    # Use a runtime_env to ensure that the agents are alive.
+    @ray.remote(runtime_env={"env_vars": {"FOO": "BAR"}})
+    def f() -> Tuple[str, int]:
+        return ray.get_runtime_context().get_worker_id(), os.getpid()
 
-    # Run the basic workload.
-    @ray.remote
-    def f():
-        for i in range(10):
-            print(f"test {i}")
+    driver_id, driver_pid = (ray.get_runtime_context().get_worker_id(), os.getpid())
+    worker_id, worker_pid = ray.get(f.remote())
 
-    # Create a runtime env to make sure dashboard agent is alive.
-    ray.get(f.options(runtime_env={"env_vars": {"A": "a", "B": "b"}}).remote())
+    python_core_driver_filename = (
+        PROCESS_TYPE_PYTHON_CORE_WORKER_DRIVER + f"-{driver_id}_{driver_pid}"
+    )
+    python_core_worker_filename = (
+        PROCESS_TYPE_PYTHON_CORE_WORKER + f"-{worker_id}_{worker_pid}"
+    )
+    job_id = ray.get_runtime_context().get_job_id()
+    worker_filename = PROCESS_TYPE_WORKER + f"-{worker_id}-{job_id}-{worker_pid}"
+
+    component_to_suffixes = [
+        (PROCESS_TYPE_DASHBOARD, [".log", ".out", ".err"]),
+        (PROCESS_TYPE_DASHBOARD_AGENT, [".log"]),
+        (PROCESS_TYPE_GCS_SERVER, [".out", ".err"]),
+        (PROCESS_TYPE_LOG_MONITOR, [".log", ".err"]),
+        (PROCESS_TYPE_MONITOR, [".log", ".out", ".err"]),
+        (PROCESS_TYPE_RAYLET, [".out", ".err"]),
+        (PROCESS_TYPE_RUNTIME_ENV_AGENT, [".log", ".out", ".err"]),
+        (python_core_driver_filename, [".log"]),
+        (python_core_worker_filename, [".log"]),
+        (worker_filename, [".out", ".err"]),
+    ]
 
     paths = list(log_dir_path.iterdir())
 
-    def component_and_suffix_exists(component, paths):
-        component, suffixes = component
+    def _assert_component_logs_exist(
+        paths: List[str], component_name: str, suffixes: List[str]
+    ):
+        suffixes_to_find = deepcopy(suffixes)
         for path in paths:
-            filename = path.stem
-            suffix = path.suffix
-            if component in filename:
```

Comment:
there was a bug here: some of the components share prefixes (like `dashboard` and `dashboard_agent`)